kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/BaseOperation.java

```diff
@@ -954,8 +954,34 @@ public Boolean isReady() {
     return i instanceof HasMetadata && Readiness.isReady((HasMetadata)i);
   }
 
+  protected T waitUntilExists(long timeoutInMillis) {
+    long end = System.currentTimeMillis() + timeoutInMillis;
+    while (System.currentTimeMillis() < end) {
+      T item = get();
+      if (item != null) {
+        return item;
+      }
+
+      // TODO is there a better way?
```

Comment:
While implementing the `waitUntilReady` I found that the optimal approach is to go hybrid, by combining a while/wait loop and watchers. This limits the number of request send to the server, but also addresses potential races that can affect a purely watcher approach.

For me going with a pure while/wait just for the exists check is good enough at least for now.